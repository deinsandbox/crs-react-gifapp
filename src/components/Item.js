import React from "react";
import PropTypes from "prop-types";

const Item = ({ url, title }) => (
  <img
    className="masonry-item animate__animated animate__fadeIn animate__delay-1s"
    src={url}
    alt={title}
  />
);

Item.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default Item;
