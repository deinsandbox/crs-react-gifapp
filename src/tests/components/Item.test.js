import React from "react";
import { shallow } from "enzyme";
import Item from "../../components/Item";

describe("Test on <Item /> component", () => {
  const url = "http://domine/image.png";
  const title = "Image description";
  const wrapper = shallow(<Item url={url} title={title} />);

  test("should render the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have an url on image source", () => {
    const image = wrapper.find("img");
    // console.log(JSON.stringify(image.html()));
    // console.log(JSON.stringify(image.props()));
    // console.log(JSON.stringify(image.prop("src")));
    expect(image.prop("src")).toBe(url);
    expect(image.prop("alt")).toBe(title);
  });

  test("should have fade_in class animation", () => {
    const figure = wrapper.find("figure");
    //console.log(JSON.stringify(figure.prop("className")));
    expect(figure.hasClass("animate__fadeIn")).toBeTruthy();
  });
});
