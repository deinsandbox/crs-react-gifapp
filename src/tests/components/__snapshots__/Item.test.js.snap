// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Item /> component should render the component correctly 1`] = `
<img
  alt="Image description"
  className="masonry-item animate__animated animate__fadeIn animate__delay-1s"
  src="http://domine/image.png"
/>
`;
